### PR TITLE
Next/main

### PR DESCRIPTION
**Pull request by jkool702**
## Summary by Sourcery

Update timep to version 1.10.1 with improved handling of BASH_ENV and expanded test coverage for complex bash behaviors.

New Features:
- Document the ability to exclude sections of code from profiling by running them under env -i.

Bug Fixes:
- Guard and preserve the BASH_ENV environment variable to avoid clobbering user configuration and handle readonly or custom values safely.

Enhancements:
- Restore the user's original BASH_ENV after timep setup completes to minimize side effects on the shell environment.

Documentation:
- Bump documented current version to 1.10.1 and describe the new BASH_ENV guard, stress test, and env -i profiling behavior.
- Clarify README flag output formatting for the -o option.

Tests:
- Add a new stress test script exercising edge cases such as env -i, builtin trap usage, exec -c, aggressive file descriptor closing, readonly BASH_ENV, and eval-based dynamic code, along with corresponding golden flamegraph and profile outputs.

### CODE REVIEW

File: TESTS/gemini-3.0/test.gemini-3.0.bash [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    ) & wait $!
+
+
+    # ---------------------------------------------------------
+    # ATTACK 5: The "Readonly BASH_ENV" Defense
+    # ---------------------------------------------------------
+    # Instead of unsetting it, we make BASH_ENV readonly.
+    # If timep tries to overwrite BASH_ENV to inject itself, 
+    # bash will throw an error and potentially crash the trap handler.
+    echo ">> Attack 5: Readonly BASH_ENV"
```
[the remark this answers is not in the document]
Comment (jkool702) [reply TESTS/gemini-3.0/test.gemini-3.0.bash:103]: > After timep runs, that behavior remains in effect.

the code timep is profiling is added to a script, and then that script is called using a new bash instance. all the changes made by the script (including to environment variables) should never persist after the script ends.
